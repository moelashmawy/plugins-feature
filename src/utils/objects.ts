export const getKeyByPropName = (
  data: any,
  keyToFind: string,
  prop: string
): string => {
  const tabName = Object.keys(data).find(
    (tabName) => data[tabName][prop] === keyToFind
  )!;
  return tabName;
};
