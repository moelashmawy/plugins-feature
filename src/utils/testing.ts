import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { expect } from 'vitest'
import { configureStore } from '@reduxjs/toolkit';
import pluginsSLice from '../state/slices/pluginsSLice';

export const createSnapshot = (ComponentToTest: ReactElement) => {
    const { asFragment } = render(ComponentToTest);
    expect(asFragment()).toMatchSnapshot();
}

export const createTestStore = () => {
  const store = configureStore(
    {
      reducer: {
        plugins: pluginsSLice
      }
    }
  );
  return store;
}