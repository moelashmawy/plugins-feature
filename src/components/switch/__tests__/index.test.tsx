import { it } from 'vitest'
import { Switch } from '..';
import { createSnapshot } from '../../../utils/testing';

it('should render checked switch', () => {
  createSnapshot(<Switch isChecked={true} onChange={() => {}} />);
});

it('should render unchecked switch', () => {
  createSnapshot(<Switch isChecked={false} onChange={() => {}} />);
});