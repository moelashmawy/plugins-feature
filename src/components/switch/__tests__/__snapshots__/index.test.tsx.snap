// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render checked switch 1`] = `
<DocumentFragment>
  <label
    class="switch"
  >
    <input
      checked=""
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</DocumentFragment>
`;

exports[`should render unchecked switch 1`] = `
<DocumentFragment>
  <label
    class="switch"
  >
    <input
      type="checkbox"
    />
    <span
      class="slider round"
    />
  </label>
</DocumentFragment>
`;
